src/hooks.js

```diff
@@ -0,0 +1,246 @@
+      if (result) {
+        return createOrUpdateHook(aHook, config);
+      }
+      throw new Parse.Error(143,`class ${aHook.className} does not exist`); 
+    });
+  }
+};
+
+var handlePost = function(req) {
+  return createHook(req.body, req.config);
+};
+
+var handleGetFunctions = function(req) {
+    var hooksDB = req.config.hooksStorageController;
+    if (req.params.functionName) {
+      return hooksDB.getFunction(req.params.functionName).then( (foundFunction) => {
+        if (!foundFunction) {
+          throw new Parse.Error(143, `no function named: ${req.params.functionName} is defined`);
+        }
+        return Promise.resolve({response: foundFunction});
+      });
+    }
+    
+    return hooksDB.getFunctions().then((functions) => {
+     return { response: functions || [] };
+    }, (err) => {
+      throw err;
+    });
+}
+
+var handleGetTriggers = function(req) {
+    var hooksDB = req.config.hooksStorageController;
+    if (req.params.className && req.params.triggerName) {
+      
+      return hooksDB.getTrigger(req.params.className, req.params.triggerName).then((foundTrigger) => {
+        if (!foundTrigger) {
+          throw new Parse.Error(143,`class ${req.params.className} does not exist`);
+        }
+        return Promise.resolve({response: foundTrigger});
+      });
+    }
+    
+    return hooksDB.getTriggers().then((triggers) => ({ response: triggers || [] }));
+}
+
+var handleDelete = function(req) {
+  var hooksDB = req.config.hooksStorageController;
+  if (req.params.functionName) {
+    triggers.removeFunction(req.params.functionName, req.config.applicationId);
+    return hooksDB.deleteFunction(req.params.functionName).then(() => ({response: {}}))
+
+  } else if (req.params.className && req.params.triggerName) {
+     triggers.removeTrigger(req.params.triggerName, req.params.className,req.config.applicationId);
+     return hooksDB.deleteTrigger(req.params.className, req.params.triggerName).then(() => ({response: {}}))
+  }
+  return Promise.resolve({response: {}});
+}
+
+var handleUpdate = function(req) {
+  var hook;
+  if (req.params.functionName && req.body.url) {
+    hook = {}
+    hook.functionName = req.params.functionName;
+    hook.url = req.body.url;
+  } else if (req.params.className && req.params.triggerName && req.body.url) {
+    hook = {}
+    hook.className = req.params.className;
+    hook.triggerName = req.params.triggerName;
+    hook.url = req.body.url
+  } else {
+    throw new Parse.Error(143, "invalid hook declaration");
+  } 
+  return updateHook(hook, req.config);
+}
+
+var handlePut = function(req) {
+  var body = req.body;
+  if (body.__op == "Delete") {
+    return handleDelete(req);
+   } else {
+  	return handleUpdate(req);
+  }
+}
+
+function enforceMasterKeyAccess(req) {
+  return new Promise((fulfill, reject) => {
+    if (!req.auth.isMaster) {
+      console.log("REQ AUTH IS NOT MASTER!");
```

Comment:
Lets not log stuff via console.log. We probably don't need to log anything here, if we do we should log it via the LoggerAdapter.